operator/builtin/transformer/add/add_test.go

```diff
@@ -0,0 +1,270 @@
+					"new": map[interface{}]interface{}{
+						"nest": map[interface{}]interface{}{
+							"key": "val",
+						},
+					},
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"add_attribute",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewAttributeField("new")
+				cfg.Value = "newVal"
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"new": "newVal"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_resource",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = "newVal"
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Resource = map[string]string{"new": "newVal"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_resource_expr",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = `EXPR($.key + "_suffix")`
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Resource = map[string]string{"new": "val_suffix"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_int_to_body",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewBodyField("new")
+				cfg.Value = 1
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": 1,
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"add_array_to_body",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewBodyField("new")
+				cfg.Value = []int{1, 2, 3, 4}
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": []int{1, 2, 3, 4},
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"overwrite",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewBodyField("key")
+				cfg.Value = []int{1, 2, 3, 4}
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": []int{1, 2, 3, 4},
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"add_int_to_resource",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = 1
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": 1,
```

Comment:
Since this test is expected to fail, it's odd that the value ended up on the body. Seems like the expected output entry should be the same as the input.